test/dotnet-vstest.Tests/VSTestTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli.VSTest.Tests
+{
+    public class VSTestTests: TestBase
+    {
+        [Fact]
+        public void TestsFromAGivenContainerShouldRunWithExpectedOutput()
+        {
+            // Copy DotNetCoreTestProject project in output directory of project dotnet-vstest.Tests
+            string testAppName = "DotNetCoreTestProject";
+            TestInstance testInstance = TestAssetsManager.CreateTestInstance(testAppName).WithLockFiles();
+
+            string testProjectDirectory = testInstance.TestRoot;
+
+            // Build project DotNetCoreTestProject
+            new Build3Command()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should()
+                .Pass();
+
+            // Prepare args to send vstest
+            string configuration = Environment.GetEnvironmentVariable("CONFIGURATION") ?? "Debug";
+            string testAdapterPath = Path.Combine(testProjectDirectory, "bin", configuration, "netcoreapp1.0");
+            string outputDll = Path.Combine(testAdapterPath, $"{testAppName}.dll");
+            string argsForVstest = string.Concat("\"", outputDll, "\"", " --TestAdapterPath:", "\"", testAdapterPath, "\"", " --Platform:x64");
```

Comment:
`vstest` is an advanced command and directly invokes the console test runner. In this case user already has built a test assembly and directly wants to run it. May be in a different machine than where the assembly is built. This limits the ways we can reason about user intention, I do agree that we should simplify the commandline wherever possible.

> Is the plan that the test target will be able to put all this together and invoke the tool for me then?
The `test3` verb will do this for the user. It's behavior will be very similar to that of `test` today.

> Also, say, for CI, I will have to find the adapter (isn't it commonly binplaced with the app? can't you try some heuristics to find the adapter before forcing the user to find it for you?).
We have this on the backlog. The parent directory of a test assembly should be probed for adapters (in case user has done a `publish`). Unlike .net core, in the desktop world, the adapter remains common for all the test assemblies user has specified to be run in a session (so the heuristic can't just pick it up from bin folder of any one test assembly, user needs to specify it).

> Why is x86 the default? I would imagine that x64 is actually the more common one these days no? And why need to specify it at all?
I think we can get away with platform argument. User chooses the architecture with the `dotnet` tool she uses to run the command. And `testhost` app is any cpu. @Faizan2304 can you please confirm?